src/startup.publish.ts

```diff
@@ -10,11 +10,23 @@ import { StaticSettingsProvider } from "./components/staticSettingsProvider";
 import { FileSystemBlobStorage } from "./components/filesystemBlobStorage";
 import { ApimPublishModule } from "./apim.publish.module";
 import { PublishingCacheModule } from "./persistence/publishingCacheModule";
+import { ISettingsProvider } from "@paperbits/common/configuration";
 
 /* Reading settings from configuration file */
-const configFile = path.resolve(__dirname, "./config.json");
-const configuration = JSON.parse(fs.readFileSync(configFile, "utf8").toString());
-const settingsProvider = new StaticSettingsProvider(configuration);
+let settingsProvider: ISettingsProvider;
+if (process.env.NODE_ENV === "staticData") {
+    settingsProvider = new StaticSettingsProvider({
+        "environment": "publishing",
+        "managementApiUrl": "https://contoso.management.azure-api.net",
+        "managementApiAccessToken": "SharedAccessSignature&1&",
+        "useHipCaptcha": false
+    });
+}
+else {
+    const configFile = path.resolve(__dirname, "./config.json");
```

Comment:
don't you need to set config files in staticData mode as well?

Reply:
It's set inside the if, but it's not a config file, but the configuration directly as an object:
 settingsProvider = new StaticSettingsProvider(**{
        "environment": "publishing",
        "managementApiUrl": "https://contoso.management.azure-api.net",
        "managementApiAccessToken": "SharedAccessSignature&1&",
        "useHipCaptcha": false
    }**);